### PR TITLE
feat: initial generation of apikeys

### PR DESCRIPTION
**Pull request by suztomo**
Example pull request created by the new-client.py https://github.com/googleapis/google-cloud-java/pull/8517

### CODE REVIEW

**Comment by suztomo** [comment]
It needs the latest shared dependencies BOM.

```
Error:  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.10.1:compile (default-compile) on project proto-google-cloud-apikeys-v2: Compilation failure: Compilation failure: 
Error:  /home/runner/work/google-cloud-java/google-cloud-java/java-apikeys/proto-google-cloud-apikeys-v2/src/main/java/com/google/api/apikeys/v2/Restrictions.java:[941,27] cannot find symbol
Error:    symbol: method parseUnknownField(com.google.protobuf.CodedInputStream,com.google.protobuf.ExtensionRegistryLite,int)
Error:  /home/runner/work/google-cloud-java/google-cloud-java/java-apikeys/proto-google-cloud-apikeys-v2/src/main/java/com/google/api/apikeys/v2/BrowserKeyRestrictions.java:[469,27] cannot find symbol
Error:    symbol: method 
```

File: java-apikeys/samples/install-without-bom/pom.xml [hunk 1]
```diff
@@ -0,0 +1,84 @@
+<?xml version='1.0' encoding='UTF-8'?>
```
Comment (meltsufin) [inline java-apikeys/samples/install-without-bom/pom.xml:1]: This file shouldn't be included along with other poms in `samples/`

Comment (suztomo) [reply java-apikeys/samples/install-without-bom/pom.xml:1]: True.

**Comment by suztomo** [comment]
The version bump of the shared dependencies BOM in synthtool https://github.com/googleapis/synthtool/pull/1641

File: google-cloud-gapic-bom/pom.xml [hunk 2]
```diff
@@ -716,7 +716,7 @@
         <version>2.4.4-SNAPSHOT</version><!-- {x-version-update:grafeas:current} -->
       </dependency>
       <dependency>
-          <groupId>com.google.api</groupId>
+          <groupId>com.google.cloud</groupId>
```
Comment (suztomo) [inline google-cloud-gapic-bom/pom.xml:719]: The group ID was wrong.

**Comment by suztomo** [comment]
For Release Please config, https://github.com/googleapis/google-cloud-java/pull/8480 needs to happen first.

**Comment by suztomo** [comment]
https://github.com/googleapis/google-cloud-java/pull/8480 is merged. Regenerating this pull request.

File: google-cloud-gapic-bom/pom.xml [hunk 3]
```diff
@@ -716,7 +716,7 @@
...
         <artifactId>google-cloud-notification</artifactId>
```
Comment (suztomo) [inline google-cloud-gapic-bom/pom.xml:720]: Why this change?
